src/planscape/forsys/merge_polygons_test.py

```diff
@@ -0,0 +1,163 @@
+        self.assertEqual(poly.geom_type, "Polygon")
+        self.assertEqual(len(poly.coords), 1)
+        self.assertEqual(self._coords_to_pixel_coords(poly.coords[0]),
+                         ((5, 5), (0, 5), (0, 0), (8, 0),
+                          (8, 3), (5, 3), (5, 5)))
+
+    def test_merges_three_nonoverlapping_polygons(self) -> None:
+        p1 = self._create_polygon(((0, 0), (0, 5), (5, 5), (5, 0), (0, 0)))
+        # p2 is not contiguous or overlapping with p1.
+        p2 = self._create_polygon(((6, 0), (6, 3), (8, 3), (8, 0), (6, 0)))
+        # p3 is contiguous with p1 and not contiguous or overlapping with p2.
+        p3 = self._create_polygon(((0, 0), (0, 1), (-1, 1), (-1, 0), (0, 0)))
+        poly = merge_polygons([p1, p2, p3], self.margin)
+        self.assertEqual(poly.srid, self.srid)
+        self.assertEqual(poly.geom_type, "MultiPolygon")
+        self.assertEqual(len(poly.coords), 2)
+        self.assertEqual(self._coords_to_pixel_coords(poly.coords[0][0]),
+                         ((5, 0), (5, 5), (0, 5), (0, 1), (-1, 0), (5, 0)))
+        self.assertEqual(self._coords_to_pixel_coords(poly.coords[1][0]),
+                         ((8, 0), (8, 3), (6, 3), (6, 0), (8, 0)))
+
+    def test_merges_three_overlapping_polygons(self) -> None:
+        p1 = self._create_polygon(((0, 0), (0, 5), (5, 5), (5, 0), (0, 0)))
+        # p2 is not contiguous or overlapping with p1.
+        p2 = self._create_polygon(((6, 0), (6, 3), (8, 3), (8, 0), (6, 0)))
+        # p3 is contiguous with p1 and p2.
+        p3 = self._create_polygon(((5, 0), (5, 5), (6, 5), (6, 0), (5, 0)))
+        poly = merge_polygons([p1, p2, p3], self.margin)
+        self.assertEqual(poly.srid, self.srid)
+        self.assertEqual(poly.geom_type, "Polygon")
+        self.assertEqual(len(poly.coords), 1)
+        self.assertEqual(self._coords_to_pixel_coords(poly.coords[0]),
+                         ((0, 5), (0, 0), (8, 0), (8, 3),
+                          (6, 3), (6, 5), (0, 5)))
+
+    def test_merges_overlapping_polygons_torus(self) -> None:
+        p1 = self._create_polygon(((0, 0), (0, 5), (1, 5), (1, 0), (0, 0)))
+        p2 = self._create_polygon(((0, 0), (0, 1), (5, 1), (5, 0), (0, 0)))
+        p3 = self._create_polygon(((4, 0), (4, 5), (5, 5), (5, 0), (4, 0)))
+        p4 = self._create_polygon(((0, 4), (0, 5), (5, 5), (5, 4), (0, 4)))
+        poly = merge_polygons([p1, p2, p3, p4], self.margin)
+        self.assertEqual(poly.srid, self.srid)
+        self.assertEqual(poly.geom_type, "Polygon")
+        self.assertEqual(poly.num_interior_rings, 1)
+        self.assertEqual(len(poly.coords), 2)
+        self.assertEqual(self._coords_to_pixel_coords(poly.coords[0]),
+                         ((4, 0), (5, 0), (5, 5), (0, 5), (0, 0), (4, 0)))
+        self.assertEqual(self._coords_to_pixel_coords(poly.coords[1]),
+                         ((4, 1), (1, 1), (1, 4), (4, 4), (4, 1)))
+
+    def test_merges_polygons_along_diagonal(self) -> None:
```

Comment:
what does it mean to merge along diagonal? it seems that these two aren't getting merged at all. is that expected?

Reply:
yup - that's expected since it seems like polygons that share a single point are not merged. I'll add a comment for this.